tests/SimpleDecodeDotProto.hs

```diff
@@ -7,15 +7,44 @@ module Main where
 
 import Test.Tasty
 import Test.Tasty.HUnit (Assertion, (@?=), (@=?), testCase)
+import Control.Monad (when)
 import qualified Data.Map as M
 import Proto3.Suite
+import qualified Proto3.Suite.JSONPB as JSONPB
+import qualified Data.Bifunctor as Bifunctor
 import qualified Data.ByteString.Char8 as BC
+import qualified Data.ByteString.Lazy as BL
+import Data.List (intercalate)
+import GHC.Stack (HasCallStack)
+import System.Environment (lookupEnv)
+import System.Exit (die)
 import System.IO
+import System.IO.Unsafe (unsafePerformIO)
+import Text.Read (readEither)
 
 import TestProto
 import qualified TestProtoImport
 import qualified TestProtoOneof
 import qualified TestProtoOneofImport
+import qualified TestProtoWrappers
+
+data Format = Binary | Jsonpb
+  deriving (Bounded, Enum, Eq, Read, Show)
+
+-- | In order to keep using 'defaultMain', which is convenient, we take
+-- the format as an environment variable instead of a command line argument.
+format :: Format
+format = unsafePerformIO $ do
+  maybeFormat <- lookupEnv "FORMAT"
+  case maybeFormat of
+    Nothing -> die $
+      "Please export a value for the FORMAT environment variable: one of: " ++
+      (intercalate ", " (map show ([minBound .. maxBound] :: [Format])))
+    Just formatStr -> case readEither formatStr of
+      Left msg -> die $
+        "Bad value of FORMAT environment variable: " ++ show format ++
+        ": " ++ show msg
+      Right f -> pure f
```

Comment:
Since you're already using implicit parameters, I would define this in `main` instead, and pass it within the context of `tests` (Each of the terms becomes `(?format :: Format) => TestTree`).

Otherwise this gets a bit confusing to me, I had trouble understanding `testCaseInFormat`

Reply:
Changed by 03fd66aad1eb978f9fb589745d3afdd1cd5055d9; does that resolve the issue?